admin/controllers/check-form.js

```diff
@@ -0,0 +1,269 @@
+      if (checkWindows[f._id]) {
+        f.checkWindows = checkWindows[f._id].map(cw => { return cw.toJSON() })
+      } else {
+        f.checkWindows = []
+      }
+    })
+
+    req.breadcrumbs(res.locals.pageTitle)
+    return res.render('test-developer/upload-and-view-forms', {
+      forms: formData,
+      breadcrumbs: req.breadcrumbs()
+    })
+  } catch (error) {
+    return next(error)
+  }
+}
+
+/**
+ * Remove check form.
+ * @param req
+ * @param res
+ * @param next
+ * @returns {Promise.<*>}
+ */
+const removeCheckForm = async (req, res, next) => {
+  const id = req.params.formId
+  try {
+    const CheckWindow = await checkFormDataService.getActiveForm(id).exec()
+    if (!CheckWindow) {
+      return next(new Error(`Unable to find form.id [${id}]`))
+    }
+    // Un-assign check-form from any check-windows
+    const checkWindowsByForm = await checkWindowService.getCheckWindowsAssignedToForms()
+    if (checkWindowsByForm[CheckWindow._id]) {
+      // Array of CheckWindows models, each with a forms array
+      let modifiedCheckWindows = []
+      checkWindowsByForm[CheckWindow._id].forEach(cw => {
+        const index = cw.forms.indexOf(CheckWindow._id)
+        if (index > -1) {
+          cw.forms.splice(index, 1)
+          modifiedCheckWindows.push(cw)
+        }
+      })
+      // Update any changed check windows
+      const promises = modifiedCheckWindows.map(cw => { cw.save() })
+      await Promise.all(promises)
+    }
+    await checkWindowService.markAsDeleted(CheckWindow)
+  } catch (error) {
+    return next(error)
+  }
+
+  res.redirect('/test-developer/upload-and-view-forms')
+}
+
+/**
+ * Upload check form view.
+ * @param req
+ * @param res
+ * @param next
+ * @returns {Promise.<void>}
+ */
+const uploadCheckForm = async (req, res, next) => {
+  req.breadcrumbs('Upload and view forms', '/test-developer/upload-and-view-forms')
+  res.locals.pageTitle = 'Upload new form'
+  try {
+    req.breadcrumbs(res.locals.pageTitle)
+    res.render('test-developer/upload-new-form', {
+      breadcrumbs: req.breadcrumbs()
+    })
+  } catch (error) {
+    next(error)
+  }
+}
+
+/**
+ * Save check form (POST).
+ * @param req
+ * @param res
+ * @param next
+ * @returns {Promise.<*>}
+ */
+const saveCheckForm = async (req, res, next) => {
+  let uploadError = {}
+  let uploadFile = req.files.csvFile
+  let checkForm = new CheckForm()
+  let absFile
+  let deleteDir
+  let forms
+  let formData
+
+  // Various errors cause a page to be rendered instead, and it *needs* a title
+  res.locals.pageTitle = 'Upload check form'
+  req.breadcrumbs(res.locals.pageTitle)
+
+  // Pick up the list of forms in case we have an error and need to re-render the page
+  try {
+    forms = await checkFormDataService.getActiveForms().sort({createdAt: -1}).exec()
+    formData = forms.map(e => { return e.toJSON() })
+    const checkWindows = await checkWindowService.getCheckWindowsAssignedToForms()
+    formData.forEach(f => {
```

Comment:
same as above.

Reply:
Same reply, this PR is only the correct routing and basic pages, more code refactoring comes next -or else this PR will become way too big.